s4584245-Yue-GCN

yuexiangzh opened the pull request:
Hi Shakes,
This pull request contains the files relating to implementing the GCN task. The descriptions of the files are below:
GCN_Layer.py: Consrtucting the two GCN layers.
load_data.py: Process the data-set.
test_driver_script.py: It seems like the main function in this project.
tsne.png：A TSNE embeddings plot with ground truth in colors.
acc.png: The line chart of Validation accuracy.
gcn.png: It would be used in README.md file.
README.md：The introduction of project and result(accuracy&tsne) of GCNs
Download the files in the same document and run the test_driver_script.py.:)

Best Regards,
YueXiang Zhao
s4584245

Commit:
First of all, I have to load and normalize the data-set.

Commit:
add some details

Commit:
Finish the load data function. I also print the shape of feature,edges
and target.

Commit:
Upload the GCN layers class.

Commit:
Input some images in the readme.md

Commit:
upload the GCN train model.

Commit:
Fix some bugs in GCN_train python file.

Commit:
update and fix some bug(variable name) in load_data.py

Commit:
Add some details in test.py and update it. :)

Commit:
the plot of accuracy

Commit:
update the acc.png

Commit:
update tsne.png

Commit:
update the load_data.py

Commit:
update gcn train file.

Commit:
update the GCN_layer file. :)

Commit:
upload the README.mupload the README.mdd

Commit:
update the gcn layer and add notes.

Commit:
add notes in gcn train

yuexiangzh committed:
add notes

yuexiangzh committed:
fix bug in train.py

gayanku commented:
- Last commit on Nov 1. Commits span 4 days(Oct 28 to Nov 1)
- Commit comments could be more descriptive.
- Readme is well done with background, results, diagrams.
- multi layer (2) layer GCN, 
**- invalid for due to code being incomplete and not matching provided results.**

- Reported a _val_ accuracy (+80%). This is however not from the code in the PR.
- Unsure if the code runs. test_driver_script.py calls a non-existent method train()
- Low code structure and modularization. Avoid the use of global variables. For example tsne() 
- Has used fully supervised split (10000/2000/3000). In future make sure that n_train < n_test for Semi supervised.

yuexiangzh commented:
> * 最后一次提交时间为 11 月 1 日。提交时间跨度为 4 天（10 月 28 日至 11 月 1 日）
> * 提交评论可能更具描述性。
> * 自述文件在背景、结果、图表方面做得很好。
> * 多层 (2) 层 GCN，
>   **- 由于代码不完整且与提供的结果不匹配而无效。**
> * 报告了_val_准确度 (+80%)。然而，这不是来自 PR 中的代码。
> * 不确定代码是否运行。test_driver_script.py 调用了一个不存在的方法 train()
> * 低代码结构和模块化。避免使用全局变量。例如 tsne()
> * 已使用完全监督拆分 (10000/2000/3000)。将来确保 n_train < n_test 用于半监督。

yuexiangzh committed:
I have fix the bug test_driver_script.py. The reason why it does not run is that I did not clear the Class Train().

yuexiangzh commented:
Hello gayanku，thanks for your comments, I have fixed the bugs in the test_drive_script.I did not clear the Class Train() so that you cannot call method train.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Gayan ***@***.***>
Sent: Wednesday, November 17, 2021 11:25:05 AM
To: shakes76/PatternFlow ***@***.***>
Cc: Yuexiang Zhao ***@***.***>; Author ***@***.***>
Subject: Re: [shakes76/PatternFlow] s4584245-Yue-GCN (PR #223)


  *   Last commit on Nov 1. Commits span 4 days(Oct 28 to Nov 1)

  *   Commit comments could be more descriptive.

  *   Readme is well done with background, results, diagrams.

  *   multi layer (2) layer GCN,
- invalid for due to code being incomplete and not matching provided results.

  *   Reported a val accuracy (+80%). This is however not from the code in the PR.

  *   Unsure if the code runs. test_driver_script.py calls a non-existent method train()

  *   Low code structure and modularization. Avoid the use of global variables. For example tsne()

  *   Has used fully supervised split (10000/2000/3000). In future make sure that n_train < n_test for Semi supervised.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/shakes76/PatternFlow/pull/223#issuecomment-971030407>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AV5N5YWNF76R7RUYANVPJALUML77DANCNFSM5G4XC5PA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

yuexiangzh committed:
Update GCN_Layer.py

yuexiangzh committed:
Update test_driver_script.py

yuexiangzh committed:
split the dataset.

yuexiangzh committed:
Split the data

Spliting the data-set into three parts, the number of data:
1. train:2000
2. valid:2000
3.test:8000

yuexiangzh commented:
I can run the  test_drive_script.py in my pc, but I did not publish the latest version of code. Now I have fixed all bugs and republished it to the pull request.
1. Spliting the dataset into three parts (test:10000/valid:2000/train:3000) , so that n_train < n_test because of Semi supervised(load_data.py)
2. Fixed the naming error in GCN_layer.py
3. Clear Train() class in test_driver_script.py
4. Method train() has defined in GCN_train.py(Class Train).
As a result, the val_accuracy will reach at about 88%. Moveover, I will update the result and README.md.

yuexiangzh commented:
You are able to run the test_driver_script.py now, here is a screenshot of the running. :)

![U$BOVE96SH}NHO3ST@E7ACJ](https://user-images.githubusercontent.com/91938530/142751751-8d967da6-bc72-4907-be52-c5c5a5e13b37.png)

shakes76 commented:
## TF/Torch Usage
Code was not running at time of submission was due -5
Solves the problem now working but changes after due date -3
See comments above from tutor
## Good Practice (Design/Commenting)
Good spacing and comments
No Header blocks -1
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present but not working on time -1
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages, could be more descriptive -1
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch)
The code did not work and will not be merged, but effort made to get it working, so feedback marks received.
Request Description OK
